client/js/main.js

```diff
@@ -10,6 +10,10 @@ const dropTargets = new Map();
 function getValidDropTargets(widget) {
   const targets = [];
   for(const [ _, t ] of dropTargets) {
+    if(t.p('dropLimit') > -1 && t.p('dropLimit') <= t.children().length)
+      if(t.p('dropLimit') < t.children().length || t.children().indexOf(widget) == -1)
```

Comment:
Could you add a comment explaining the logic? I don't understand the reason why we don't the widget be droppable back onto its existing parent even if it's reached its limit. Wouldn't that create a situation where as soon as a you pick up a card out of a holder, you can't put it back if its limit was changed while holding it?

```gherkin
Scenario Outline: drop limit changed
  Given a holder with drop limit <limit_start>
  And it has <num_children> children
  And its drop limit is changed to <limit_new>
  When I pick up a child out of the holder
  Then I <can_or_cannot> drop it back down on the holder
Examples:
  | limit_start | num_children | limit_new | can_or_cannot |
  | -1          | 3            | 3         | can           |
  | -1          | 4            | 3         | TBD           |
```

Reply:
Do those help?

For your second example I would assume it can't be dropped (should be the current behaviour). The first example is what the second if is for (makes it `can`).

Reply:
Yes, it helps, but my point is I think that there's a use case for letting the child be dropped back onto its over-the-limit holder:
* Holder has 4 cards
* I pick a card up, but now I can't drop it back down into the holder
* But I made a mistake and meant to click a different pile, but now the game is ruined and have to start over.

Reply:
Just to clarify: is the drop limit considered both for player's drag actions and also for MOVE operations? It could be quite useful for MOVE, for instance when you want to deal cards up to the maximum for the holder.

Reply:
At the moment only player's drag. Not sure if `MOVE` should use the same property.

Reply:
It should not limit MOVE imho

Reply:
I changed the behaviour so a widget can always be dropped back into the same holder. I think the `MOVE` update should be its own PR.